server-common/src/main/java/org/apache/kafka/server/common/Feature.java

```diff
@@ -0,0 +1,349 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import org.apache.kafka.common.feature.SupportedVersionRange;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV0_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV0_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV1_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV1_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV2_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV2_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV3_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV3_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV4_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV4_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV5_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV5_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV6_0;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV6_1;
+import static org.apache.kafka.server.common.TestFeatureVersion.UT_FV7_0;
+
+/**
+ * This is enum for the various features implemented for Kafka clusters.
+ * KIP-584: Versioning Scheme for Features introduced the idea of various features, but only added one feature -- MetadataVersion.
+ * KIP-1022: Formatting and Updating Features allowed for more features to be added. In order to set and update features,
+ * they need to be specified via the StorageTool or FeatureCommand tools.
+ * <br>
+ * Having a unified enum for the features that will use a shared type in the API used to set and update them
+ * makes it easier to process these features.
+ */
+public enum Feature {
+
+    /**
+     * Features defined. If a feature is included in this list, and marked to be used in production they will also be specified when
+     * formatting a cluster via the StorageTool. MetadataVersion is handled separately, so it is not included here.
+     *
+     * See {@link TestFeatureVersion} as an example. See {@link FeatureVersion} when implementing a new feature.
+     */
+    TEST_VERSION("test.feature.version", TestFeatureVersion.valuesWithoutUnitTestVersions(), TestFeatureVersion.LATEST_PRODUCTION),
```

Comment:
This is an existing issue. Could we use the predefined constant for "test.feature.version"? Ditto in other features.